json-schema-core/src/main/java/io/ballerina/jsonschema/core/GeneratorUtils.java

```diff
@@ -114,180 +115,223 @@ public class GeneratorUtils {
-            return NUMBER;
+    static void processRequiredFields(Map<String, RecordField> recordFields) {
+        boolean changeFlag = true;
+        while (changeFlag) {
+            changeFlag = false;
+            for (Map.Entry<String, RecordField> entry : recordFields.entrySet()) {
+                String key = entry.getKey();
+                RecordField keyRecord = recordFields.get(key);
+                RecordField value = entry.getValue();
+
+                if (value.getDependentRequired() == null || !keyRecord.isRequired()) {
+                    continue;
+                }
+
+                // If there are dependentRequired items iterate through them
+                for (String dependentRequired : value.getDependentRequired()) {
+                    if (!recordFields.get(dependentRequired).isRequired()) {
+                        recordFields.get(dependentRequired).setRequired();
+                        changeFlag = true;
+                    }
+                }
+            }
         }
+    }
 
-        if (invalidLimits(minimum, exclusiveMinimum, maximum, exclusiveMaximum)) {
-            return NEVER;
+    // Create a new type with the union of typedesc and return the new type name.
+    static String resolveTypeNameForTypedesc(String name, String typeName, Generator generator) {
+        if (!typeName.contains(PIPE)) {
+            return typeName;
         }
+        String newType = resolveNameConflicts(name, generator);
+        String typeDeclaration = PUBLIC + WHITE_SPACE + TYPE + WHITE_SPACE + newType + WHITE_SPACE + typeName +
+                SEMI_COLON;
+        ModuleMemberDeclarationNode moduleNode = NodeParser.parseModuleMemberDeclaration(typeDeclaration);
+        generator.nodes.put(newType, moduleNode);
+        return newType;
+    }
 
-        generator.addImports(BAL_JSON_DATA_MODULE);
-        String finalType = resolveNameConflicts(convertToPascalCase(name), generator);
+    static ArrayList<String> processRecordFields(Map<String, RecordField> recordFields, Generator generator) {
+        ArrayList<String> recordBody = new ArrayList<>();
 
-        List<String> annotationParts = new ArrayList<>();
+        for (Map.Entry<String, RecordField> entry : recordFields.entrySet()) {
+            String key = entry.getKey();
+            RecordField value = entry.getValue();
 
-        addIfNotNull(annotationParts, MINIMUM, minimum);
-        addIfNotNull(annotationParts, EXCLUSIVE_MINIMUM, exclusiveMinimum);
-        addIfNotNull(annotationParts, MAXIMUM, maximum);
-        addIfNotNull(annotationParts, EXCLUSIVE_MAXIMUM, exclusiveMaximum);
-        addIfNotNull(annotationParts, MULTIPLE_OF, multipleOf);
+            ArrayList<String> fieldAnnotation = new ArrayList<>();
 
-        String formattedAnnotation = getFormattedAnnotation(annotationParts, NUMBER_ANNOTATION, finalType, NUMBER);
+            if (value.getDependentSchema() != null) {
```

Comment:
Extract values/fields that can be repeatedly accessed to variables. E.g., - `value.getDependentSchema()`